lib/elixir/lib/application.ex

```diff
@@ -4,32 +4,90 @@ defmodule Application do
+  resource file:
+
+      iex(1)> Application.load(:ex_unit)
+      :ok
+
+  If an application has included applications, they are loaded too. And the
+  procedure recurses if they in turn have included applications. (Included
+  applications are unrelated to applications in Mix umbrella projects, they are
+  an Erlang concept that has to do with coordinated starts.)
+
+  When an application is loaded, the environment specified in its resource file
+  is merged with any overrides from config files passed to `erl` via the
+  `-config` option. It is worth highlighting that releases pass `sys.config`
+  this way. In turn, these are merged again with any specific `-Application`
+  flags passed to `erl`.
+
+  Loading an application _does not_ load its modules.
+
+  In practice, you rarely load applications by hand because that is part of the
+  start process, explained next.
+
+  ## Starting and stopping applications
+
+  Applications are also _started_, and _stopped_:
+
+      iex(1)> Application.start(:ex_unit)
+      :ok
+      iex(2)> Application.stop(:ex_unit)
+      :ok
 
   Once your application is compiled, running your system is a matter of starting
   your current application and its dependencies. Differently from other languages,
   Elixir does not have a `main` procedure that is responsible for starting your
   system. Instead, you start one or more applications, each with their own
   initialization and termination logic.
 
-  Applications also provide an "application environment", which provides one
-  mechanism for configuring long running applications. We will learn more about
-  the tooling, start and shutdown and the application environment in the next
-  sections.
+  When an application is started, the BEAM loads it if it is not already loaded
+  (in the technical sense described above). Then, it checks if the dependencies
+  listed in the `applications` key of the resource file are already started.
+  Having at least one dependency not started is an error condition.
+
+  When you start an application with `mix run`, Mix takes care of starting all
+  dependencies for you.
 
-  ## Start and shutdown
+  The key `mod` of the application resource file specifies whether the
+  application has a callback module. If it doesn't, the start process is done,
+  otherwise, it continues.
 
-  Starting an application is done via the "application module callback", which
-  is a module that defines the `start/2` function. The `start/2` function should
-  then start a supervisor, which is often called as the top-level supervisor, since
-  it sits at the root of a potentially long supervision tree. When the system is
+  An application callback module is a module that defines at least the callbacks
+  `start/2` and `stop/1`. If there is one configured, the start process invokes
+  its `start/2` function, which must start and link a supervisor, and return its
+  pid. That supervisor is often called _the top-level supervisor_, since it sits
```

Comment:
I write PID everywhere except in Erlang/Elixir, where I see "pid" is more common in docs and books.

Reply:
Oh man I just saw https://github.com/eksperimental/elixir_inconsistencies, that is golden!

Reply:
We use PID in documenration since its an acronym, but we use pid for arguments and function names

Refs
https://github.com/elixir-lang/elixir/pull/4883
https://github.com/elixir-lang/elixir/pull/4754

Reply:
Awesome, I've revised it.